lightningd/lightningd.c

```diff
@@ -851,6 +851,11 @@ int main(int argc, char *argv[])
 	if (ld->config.keypass)
 		sodium_munlock(ld->config.keypass->data, sizeof(ld->config.keypass->data));
 
+	/* for giggles, we do this here */
+	db_begin_transaction(ld->wallet->db);
+	migrate_last_tx_to_psbt(ld, ld->wallet->db);
+	db_commit_transaction(ld->wallet->db);
+
```

Comment:
Don't we have the ability to add migration functions to the big array in `db.c`? That way they'd be run only once, and we'd remember that we applied them.

Reply:
We do, yes. The problem is a dependency cycle (sort of) -- the migration code relied on HMSd to fetch the local pubkey for the funding tx but the HSM isn't initialized until after the database has been migrated, so the migration fails.

Reply:
actually i can fix this dependency.